backend/common/src/main/java/ai/verta/modeldb/common/futures/FutureExecutor.java

```diff
@@ -35,19 +41,25 @@ public static FutureExecutor newSingleThreadExecutor() {
 
   @Override
   public void execute(@NonNull Runnable r) {
+    if (otelContext != null) {
+      r = otelContext.wrap(r);
+    }
+    if (grpcContext != null) {
+      r = grpcContext.wrap(r);
+    }
+
     if (GlobalTracer.isRegistered()) {
       final var tracer = GlobalTracer.get();
       final var span = tracer.scopeManager().activeSpan();
+      Runnable finalR = r;
       other.execute(
-          Context.current()
-              .wrap(
-                  () -> {
-                    try (Scope s = tracer.scopeManager().activate(span)) {
-                      r.run();
-                    }
-                  }));
+          () -> {
+            try (Scope s = tracer.scopeManager().activate(span)) {
+              finalR.run();
+            }
+          });
     } else {
-      other.execute(Context.current().wrap(r));
+      other.execute(r);
     }
```

Comment:
with OTel in place, I think we can replace all of this with just `other.execute(r)`. OpenTelemetry implements this in the tracer shim implementation itself.

Reply:
github won't let me suggest on this whole block. so, replace lines 51-63 with `other.execute(r)`

Reply:
I cleaned it up.